graph/step.go

```diff
@@ -233,39 +233,62 @@ func GetBuildCacheImageTag(taskName, stepID string) string {
 }
 
 // GetCmdWithCacheFlags adds buildx cache parameters to the cmd.
-func (s *Step) GetCmdWithCacheFlags(taskName string) (string, error) {
-	result := ""
+func (s *Step) GetCmdWithCacheFlags(taskName, registry string) (string, error) {
+	var foundDomain bool
+	var named reference.Named
 
+	if strings.ToLower(s.Cache) != enabled {
+		return "", errors.New("cache needs to be set to 'enabled' to use build cache")
+	}
```

Comment:
why do you need the check here? is it for debug purpose?

Reply:
yeah it should never really happen. But helps with unit tests